atsign-foundation/at_c
# feat: uuid library in atchops using uuid4 fork (and new ctest tooling)

closes #84 

**- What I did**

- Created a uuid.c and implemented it
- Created test_uuid.c
- Created include/uuid.h
- More tooling for atclient and atchops for ctest

**- How I did it**

- Created a fork of uuid4 for us along which include my own additions (such as support for NetBSD and OpenBSD which was taken from the PR section of the original repo) https://github.com/atsign-foundation/uuid4/releases/tag/v1.0.1
- Consumed our fork of uuid4 (with our additions)
- The fork of uuid4 only supports Windows, Apple, Linux, now openBSD and NetBSD, but not ESp32. Later on, we can update the fork to support that

**- How to verify it**

- See changes and new test passing

```
      Start 12: test_uuid4

12: Test command: /home/runner/work/at_c/at_c/packages/atchops/build/tests/test_uuid4
12: Working Directory: /home/runner/work/at_c/at_c/packages/atchops/build/tests
12: Test timeout computed to be: 2
12: (36): 090ded78-5298-447c-b2e6-0ddb70161467
12: (36): 292b6716-b889-4dde-865d-91c9586491df
12: (36): f2f64998-6f40-49f6-90eb-d4350f119b6f
12/12 Test #12: test_uuid4 .......................   Passed    0.00 sec
```

**- Description for the changelog**

```c
int atchops_uuid_init(void);
int atchops_uuid_generate(char *uuidstr, int uuidstrlen);
```
